frontend/reducers/boltCartToken.js

```diff
@@ -14,24 +15,34 @@ const boltTokenReducer = (
   state = {
     cartToken: null,
     isFetching: false,
+    willUpdate: false,
   },
   action
 ) => {
   switch (action.type) {
+    case UPDATE_PRODUCTS_IN_CART: {
```

Comment:
don't we need DELETE_PRODUCTS_FROM_CART here too?

Reply:
very good catch. I added the delete products from cart case as well as delete coupon and add coupon.